Column reorder listener blows if the grid is in multiselect mode

In Vaadin 14.1.beta1, I added a addColumnReorderListener on a grid with SelectionMode.MULTI.

I've got this stacktrace after I reordered one column from the client side: 
`
Caused by: java.lang.ClassCastException: elemental.json.impl.JreJsonNull cannot be cast to elemental.json.JsonString
	at elemental.json.impl.JreJsonArray.getString(JreJsonArray.java:108)
	at com.vaadin.flow.component.grid.ColumnReorderEvent.getSortedByIds(ColumnReorderEvent.java:95)
	at com.vaadin.flow.component.grid.ColumnReorderEvent.<init>(ColumnReorderEvent.java:61)`

I think it's caused because the columnsIDs (from "event.detail.columns") contains a JreJsonNull (the column for selection).

--- a/vaadin-grid-flow-integration-tests/src/main/java/com/vaadin/flow/component/grid/it/GridTestPage.java
+++ b/vaadin-grid-flow-integration-tests/src/main/java/com/vaadin/flow/component/grid/it/GridTestPage.java
@@ -76,6 +76,7 @@ public class GridTestPage extends Div {
 
     private void createGridWithComponentRenderers() {
         Grid<Item> grid = new Grid<>();
+        grid.setSelectionMode(Grid.SelectionMode.MULTI);
 
         AtomicBoolean usingFirstList = new AtomicBoolean(true);
 

--- a/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/GridColumnOrderHelper.java
+++ b/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/GridColumnOrderHelper.java
@@ -174,7 +174,7 @@ class GridColumnOrderHelper<T> {
 
         // holds the current immediate child columns.
         final Set<AbstractColumn<?>> childColumns = new HashSet<>();
-        column.getChildren()
+        column.getChildren().filter(c -> !(c instanceof GridSelectionColumn))
                 .forEach(it -> childColumns.add((AbstractColumn) it));
         // the new order of the children is computed here.
         final List<AbstractColumn<?>> newOrder = new ArrayList<>();

--- a/vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.js
+++ b/vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.js
@@ -955,6 +955,7 @@ window.Vaadin.Flow.gridConnector = {
 
     grid.addEventListener('column-reorder', e => {
       const columns = grid._columnTree.slice(0).pop()
+        .filter(c => c._flowId)
         .sort((b, a) => (b._order - a._order))
         .map(c => c._flowId);
 

